packages/mdc-top-app-bar/fixed/foundation.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @license
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {cssClasses} from '../constants';
+import MDCTopAppBarAdapter from '../adapter';
+import MDCTopAppBarFoundation from '../foundation';
+
+/**
+ * @extends {MDCTopAppBarFoundation<!MDCFixedTopAppBarFoundation>}
+ * @final
+ */
+class MDCFixedTopAppBarFoundation extends MDCTopAppBarFoundation {
+  /**
+   * @param {!MDCTopAppBarAdapter} adapter
+   */
+  constructor(adapter) {
+    super(adapter);
+    // State variable for the previous top app bar state
```

Comment:
Pro tip: Always use `/**` comments to document fields and methods.

Editors like IntelliJ will parse the JSDoc `/**` comments and display them inline when you hit <kbd>Ctrl</kbd>+<kbd>Q</kbd>.